### ISSUE TITLE
Add possibility to add Symfony routes to navigation

### ISSUE THREAD

**Issue by trickreich** [sulu/sulu#3051]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | yes

#### Actual Behavior

Actually you just can create a external link to a internal Symfony route.

#### Expected Behavior

You can add Symfony routes from your project to the navigation/content.

#### Possible Solutions

Mark the needed Symfony routes in the code with a special tag.
Then maybe you can add via overlay dialog a available Symfony route id.

**Comment by alexander-schranz** [comment]
@wachterjohannes I would say we should use the `LinkProviders` here in the Internal Link and create a new LinkProvider for Static Routes.

**Comment by danrot** [comment]
Not sure if this is a good idea... I would say that the content manager should not be able to remove a static route in the navigation, if it really should be there. If there is a static route you are usually programming an app, where you can fix the navigation apart from a few content pages.

**Comment by alexander-schranz** [comment]
@danrot I aggree with you that most static routes should be hardcoded because it are mostly very functional links but I also think it should be possible to link to this routes e.g.: in the ckeditor without choose an external link.

**Comment by danrot** [comment]
Yeah, you are right, the text editor might be a valid use case.

**Comment by trickreich** [comment]
How can you hardcode a static route into a sulu navigation?

**Comment by danrot** [comment]
Just use them in a twig template and render them before the `sulu_navigation_*` call.

**Comment by trickreich** [comment]
In my opinion its totally legit that a content manager decides to move a page in the navigation

**Comment by danrot** [comment]
And then might delete the link to a crucial part of the page? I'd say that risk is too high for not having a real benefit of it.

**Comment by wachterjohannes** [comment]
hmm i am not so critical as @danrot . maybe it could be "configurable" which route should be handled in the content-tree. but should be defined later.

+1 for the link-provider which would be quite easy.

**Comment by trickreich** [comment]
@danrot And when you need the navigation in a subtree?

**Comment by danrot** [comment]
You've got a point here 😄 But I am still skeptical about it.

**Comment by alexander-schranz** [comment]
I think as it is more work to make it possible for the content manager to manage it ( 1. create page, 2. tag the static route 3. link to static page, 4. maybe create a navigation context 5. add page to navigation ) as simple link to static route in the html.twig. The developer will always first do the static link for all critical navigation points. 

What we need todo to support it:

1. we need create a new content type which allows to use the LinkProviders (we did something similiar in a client project)
2. add a new link provider which search for tagged routes and make them available.

**Comment by alexander-schranz** [comment]
@sulu/core-team I renamed this issue as I think we should overthink the `Internal Type` to work with `media`, `articles` or any other custom link provider which we also use inside the text editor. For symfony routes it would also be possible to provide a [link provider](https://docs.sulu.io/en/stable/cookbook/link-provider.html) if needed.

In 1.6 @wachterjohannes in a custom project created a link provider content / field type maybe we can oriented about that it was similar like the `url` type where the dropdown was `pages`, `articles`, `media` and then an overlay did open like it is currently in the texteditor after submit overlay a target title was shown in the text field and clicking on the text field would open the overlay again.

**Comment by SirCoffeeMonkey** [comment]
+1 for enabling custom link providers

**Comment by nnatter** [comment]
Think this is now also related to https://github.com/sulu/sulu/issues/4966 🙂

**Comment by alexander-schranz** [comment]
@nnatter thx for linking this is definitly something we could also think about that we have a kind of LinkProvider for external links. Not sure how we handled this in the UX, as i should not have to need to open an overlay to paste my copied url. But I like the idea, but think we definitely need here a bigger concept how to combine DX and UX.

**Comment by nnatter** [comment]
Also related to https://github.com/sulu/sulu/issues/5790 🙂

**Comment by niklasnatter** [comment]
We finally managed to implement a universal `Link` selection in https://github.com/sulu/sulu/pull/6134 🎉 

The final step for solving this issue would be replacing the `page selection` of the `page_settings` form with the new `link` field-type.

**Comment by ElwynVdb** [comment]
> We finally managed to implement a universal `Link` selection in #6134 🎉
> 
> The final step for solving this issue would be replacing the `page selection` of the `page_settings` form with the new `link` field-type.

Is there an update on this? I'm currently facing an issue where we want to select an article to be a part of the navigation tree, but this would require for the page selection to become the new LInk field-type.

**Comment by alexander-schranz** [comment]
@ElwynVdb no changes yet. Same workarounds currently you can create a Internal Link with a External Link to a Article. This will be target in Sulu 3.0 as currently changing it would be a backwards compatibility break.

**Comment by ElwynVdb** [comment]
@alexander-schranz That's no problem, looking forwards to the 3.0 release. If i'm correct, you mean to create a page where I manually paste the link to my article using the page settings external link for now?

**Comment by alexander-schranz** [comment]
@ElwynVdb yes! We even used that on our own sites to even link Media Documents inside navigations.

**Comment by alexander-schranz** [comment]
With 3.0 it now possible to link to anything which link providers allows:

<img width="1371" height="347" alt="Image" src="https://github.com/user-attachments/assets/38d3f911-eabd-4777-9de9-7a0891542b12" />
